tests/dummy/src/resources/user/processor.ts

```diff
@@ -0,0 +1,75 @@
+import { KnexProcessor, Operation, KnexRecord } from "../../jsonapi-ts";
+import User from "./resource";
+import { HasId } from "../../../../../src/processors/operation-processor";
+
+const timeout = (data, time) => new Promise(resolve => setTimeout(() => resolve(data), time));
+
+export default class UserProcessor extends KnexProcessor<User> {
+  resourceClass = User;
+
+  async get(op: Operation): Promise<HasId[]> {
+    const { params, ref } = op;
+    const { id, type } = ref;
+    const tableName = this.typeToTableName(type);
+    const filters = params ? { id, ...(params.filter || {}) } : { id };
+    const resource = Object.create(this.resourceFor(type));
+    const fields = params ? { ...params.fields } : {};
+    const attributes = this.getColumns(
+      Object.keys(resource.__proto__.attributes),
+      fields,
+      type
+    );
+
+    let records: KnexRecord[] = await this.knex(tableName)
```

Comment:
This isn't a good loading for actual `hasMany` but I just wanted to get some basic data flowing.

Replace this with better SQL and linking of relations.

Also we should only add joins if the relation is in `include`

Reply:
The join will scope the main results, which is not a case I think so. We want to return all the Users and includes are the extend to the query, not the limitation.

Reply:
For the joins we need to do something more like https://github.com/bookshelf/bookshelf/blob/fadb11cc7a9d32fb8d72151b2a23e990e6ba95d2/lib/eager.js

That's the source code of when you want to eager load related data with bookshelf (an ORM built on top of KNEX).

Specifically how it works with `withRelated` and maps more queries.